### PR TITLE
[ubp] add chargebee cost center on cancellation

### LINKED ISSUE #15055: [UBP] Handle Chargebee cancellations
When a user cancels their chargebee plan, they should transition into a fresh "free" ubp plan, so all previous usage needs to be reset.

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Create a chargebee cost center when the plan gets cancelled. This is so that our usage component treats any accumulated usage as covered and resets to a fresh UBP free plan at the endDate.

## Related Issue(s)

Fixes #15055

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/ee/payment-endpoint/src/chargebee/subscription-handler.ts [hunk 1]
```diff
@@ -4,30 +4,30 @@
  * See License.enterprise.txt in the project root folder.
  */
```
Comment (svenefftinge) [inline components/ee/payment-endpoint/src/chargebee/subscription-handler.ts:6]: Sorry for all the formatter changes. seems like this component didn't get the big formatting update last time.

File: components/ee/payment-endpoint/src/config.ts [hunk 2]
```diff
@@ -36,11 +37,14 @@ export GITPOD_GITHUB_APP_MKT_NAME=gitpod-draft-development-app
     readonly githubAppAppID: number = process.env.GITPOD_GITHUB_APP_ID ? parseInt(process.env.GITPOD_GITHUB_APP_ID) : 0;
     readonly githubAppWebhookSecret: string = process.env.GITPOD_GITHUB_APP_WEBHOOK_SECRET || "unknown";
     readonly githubAppCertPath: string = process.env.GITPOD_GITHUB_APP_CERT_PATH || "unknown";
+    readonly usageServiceAddr: string = process.env.GITPOD_USAGE_SERVICE_ADDR || "usage.default.svc.cluster.local:9001";
 
-    readonly maxTeamSlotsOnCreation: number = !!process.env.TS_MAX_SLOTS_ON_CREATION ? parseInt(process.env.TS_MAX_SLOTS_ON_CREATION) : 1000;
+    readonly maxTeamSlotsOnCreation: number = !!process.env.TS_MAX_SLOTS_ON_CREATION
+        ? parseInt(process.env.TS_MAX_SLOTS_ON_CREATION)
```
Comment (jankeromnes) [inline components/ee/payment-endpoint/src/config.ts:43]: Nit: pre-existing unsafe `parseInt` without a radix (can accidentally parse as octal or hex in some edge cases)

```suggestion
        ? parseInt(process.env.TS_MAX_SLOTS_ON_CREATION, 10)
```

**Review by jankeromnes — COMMENTED** [review]
Thanks! Code changes look good to me.

Now I'll try to test this and make sure I can't find any bugs. 🔍

**Comment by jankeromnes** [comment]
@svenefftinge Follow-up question: What did you have in mind regarding how this PR can be tested? (I see no "How to test" instructions and no PR preview, but I'm happy to help you set up a Chargebee webhook so that we can exercise the new code path once)

**Comment by jankeromnes** [comment]
Okay, we now have a preview environment with:
- payment
- a chargebee webhook
🎉

I was able to use it to test this PR. Here is what I did:

1. Create a Team
2. Upgrade this team to "Unleashed EUR"
3. Invite a second team member

At this point, the `d_b_cost_center` looked like this:

```
mysql> select * from d_b_cost_center;
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
| id                                        | spendingLimit | deleted | _lastModified              | creationTime             | billingStrategy | nextBillingTime          | billingCycleStart        |
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
| team:deb8b396-0739-43f6-9c5e-7346e4adf8d3 |             0 |       0 | 2022-11-30 15:28:13.120132 | 2022-11-30T15:28:13.113Z | other           | 2022-12-30T15:28:13.113Z | 2022-11-30T15:28:13.113Z |
| user:189ad1d0-d345-4fe4-981d-a765be8713c3 |           500 |       0 | 2022-11-30 15:30:47.099930 | 2022-11-30T15:30:47.095Z | other           | 2022-12-30T15:30:47.095Z | 2022-11-30T15:30:47.095Z |
| user:6ed86c44-781b-40d9-89d4-5baca8edc74a |           500 |       0 | 2022-11-30 15:21:28.154066 | 2022-11-30T15:21:28.150Z | other           | 2022-12-30T15:21:28.150Z | 2022-11-30T15:21:28.150Z |
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
```

4. Enable usage-based flag for this team (waited for the flag to become active, but nothing changed in the UI)
5. Cancelled the Chargebee team plan
6. ... refreshed the Team Billing page until the cancellation came through

Eventually, the Chargebee cancellation came through, and I landed on the empty "Upgrade to pay-as-you-go" team billing page 👍

However, nothing changed in `d_b_cost_center` 🤔 is this a bug? 🐛 

```
mysql> select * from d_b_cost_center;
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
| id                                        | spendingLimit | deleted | _lastModified              | creationTime             | billingStrategy | nextBillingTime          | billingCycleStart        |
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
| team:deb8b396-0739-43f6-9c5e-7346e4adf8d3 |             0 |       0 | 2022-11-30 15:28:13.120132 | 2022-11-30T15:28:13.113Z | other           | 2022-12-30T15:28:13.113Z | 2022-11-30T15:28:13.113Z |
| user:189ad1d0-d345-4fe4-981d-a765be8713c3 |           500 |       0 | 2022-11-30 15:30:47.099930 | 2022-11-30T15:30:47.095Z | other           | 2022-12-30T15:30:47.095Z | 2022-11-30T15:30:47.095Z |
| user:6ed86c44-781b-40d9-89d4-5baca8edc74a |           500 |       0 | 2022-11-30 15:21:28.154066 | 2022-11-30T15:21:28.150Z | other           | 2022-12-30T15:21:28.150Z | 2022-11-30T15:21:28.150Z |
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
```

**Comment by jankeromnes** [comment]
Another test, this time with an individual Chargebee plan:

1. Log in as a new user
2. Upgrade to Professional EUR (individual plan)
3. Create a team called "Gitpod [Something]" (in order to enable usage-based flag for user)
4. In `/plans`, manage billing, then cancel the Professional EUR plan
5. Refresh page until the cancellation is effective
6. ... there was still no change in `d_b_cost_center`:

```
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
| id                                        | spendingLimit | deleted | _lastModified              | creationTime             | billingStrategy | nextBillingTime          | billingCycleStart        |
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
| user:0d25e4cf-d6cf-42e5-97c0-60e0e3444020 |           500 |       0 | 2022-11-30 15:59:44.365184 | 2022-11-30T15:59:44.359Z | other           | 2022-12-30T15:59:44.359Z | 2022-11-30T15:59:44.359Z |
+-------------------------------------------+---------------+---------+----------------------------+--------------------------+-----------------+--------------------------+--------------------------+
```

**Comment by svenefftinge** [comment]
We decided to handle this maually in the admin dashboard https://github.com/gitpod-io/gitpod/issues/15055
